### PR TITLE
[Major] Multi-bench setup

### PR DESCRIPTION
**Pull request by shreyasp**
Modifications to configuration files for generating multi-bench setup.

### CODE REVIEW

File: bench/tests/test_setup_production.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+				"{bench_name}-process:{ bench_name }-node-socketio[\s]*RUNNING",
+				"{bench_name}-redis:{ bench_name }-redis-cache[\s]*RUNNING",
+				"{bench_name}-redis:{bench_name}-redis-celery-broker[\s]*RUNNING",
+				"{bench_name}-redis:{bench_name}-redis-async-broker[\s]*RUNNING",
+			):
+			self.assertTrue(re.match(key.format(bench_name), out))
+
+	def assert_nginx_process(self):
+		out = bench.utils.get_cmd_output("sudo nginx -t")
+		self.assertTrue("nginx: configuration file /etc/nginx/nginx.conf test is successful" in out)
```
Comment (pdvyas) [inline bench/tests/test_setup_production.py]: Also add requests.get('http://localhost') as a final test

File: bench/config.py [hunk 2]
```diff
@@ -80,29 +108,56 @@ def generate_nginx_config(bench='.'):
...
-	conf = {
-		"port": get_config().get('redis_async_broker_port', '12311'),
-		"redis_version": get_redis_version()
-	}
-	config = template.render(**conf)
-	write_config_file(bench, 'redis_async_broker.conf', config)
+def _generate_redis_config(template_name, context, bench):
+	template = env.get_template(template_name)
+
+	if "process_files_path" not in context:
```
Comment (pdvyas) [inline bench/config.py]: Is this dir used only to hold the pid files?

Comment (shreyasp) [inline bench/config.py]: @pdvyas This holds pid files and rdb files for respective redis processes

File: bench/tests/test_setup_production.py (hunk 1 above)
Comment (shreyasp) [inline bench/tests/test_setup_production.py]: @pdvyas Point noted. This still hasn't had a sucessful run until now :+1:

File: bench/config.py (hunk 2 above)
Comment (pdvyas) [inline bench/config.py]: Ok, We need a better name for `process_files_path`

Comment (rmehta) [inline bench/config.py]: `pid_path` ?

**Comment by anandpdoshi** [comment]
- [ ] Remove `default` server block in nginx
- [ ] Remove dependence on `currentsite.txt` for nginx config generation

**Comment by anandpdoshi** [comment]
nginx config fixed!

**Pull request closed without merging**